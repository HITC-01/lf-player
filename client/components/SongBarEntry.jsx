import React from 'react';
import PropTypes from 'prop-types';
import helpers from '../helpers/playerHelpers.js';

const SongBarEntry = ({
  position, number, bar, playState, barFraction,
  handleScan, handleExit, handleClick,
}) => {
  const height = (position === 'lower') ? 30 : 70;
  let barClass = `player-songbar-${position}`;

  barClass += helpers.colorBar(barFraction, playState);
  return (
    <div
      className={barClass}
      style={{ height: Math.floor(height * bar) }}
      onClick={() => handleClick(barFraction)}
      onPointerOver={() => handleScan(number)}
    />
  );
};

SongBarEntry.propTypes = {
  position: PropTypes.string.isRequired,
  bar: PropTypes.number.isRequired,
  number: PropTypes.number.isRequired,
  playState: PropTypes.object.isRequired,
  barFraction: PropTypes.number.isRequired,
  handleScan: PropTypes.func.isRequired,
  handleExit: PropTypes.func.isRequired,
  handleClick: PropTypes.func.isRequired,
};

export default SongBarEntry;
