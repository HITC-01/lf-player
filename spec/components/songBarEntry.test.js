/* eslint-env jest */
/* eslint react/jsx-filename-extension: 0 */

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from '../enzyme';
import SongBarEntry from '../../client/components/SongBarEntry.jsx';

describe('SongBarEntry component', () => {
  let props = { };

  beforeEach(() => {
    props = {
      position: 'upper',
      barHeight: 0.5,
      playState: {
        playing: false,
        intervalId: 0,
        currentTime: 0,
        totalTime: 300,
        hoverPosition: 0,
        hovering: false,
      },
      barFraction: 0.9,
      handleScan: jest.fn(x => x),
      handleClick: jest.fn(x => x),
    };
  });

  test('check props', () => {
    const component = shallow(<SongBarEntry {...props} />);
    expect(component.props().length).toBe(4);
    expect(component.props('position')).toBe('upper');
    expect(component.props('onClick')).toBeInstanceOf(Function);
  });

  test('render single song bar upper bar with correct class', () => {
    const component = shallow(<SongBarEntry {...props} />);
    expect(component.find('div')).toBeDefined();
    expect(component.find('div').hasClass('player-songbar-upper-to-play')).toBe(true);
  });

  test('render single song bar lower bar with correct class', () => {
    props.position = 'lower';
    const component = shallow(<SongBarEntry {...props} />);
    expect(component.find('div')).toBeDefined();
    expect(component.find('div').hasClass('player-songbar-lower-to-play')).toBe(true);
  });

  test('render basic snapshot', () => {
    const tree = renderer
      .create(<SongBarEntry {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  // test('respond to clicks', () => {
  //   const component = shallow(<SongBarEntry {...props} />);
  //   component.find('div').simulate('click');
  //   expect(fraction).toBe(0.9);
  // });
});
