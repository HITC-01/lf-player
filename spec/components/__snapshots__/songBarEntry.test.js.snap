// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SongBarEntry component render basic snapshot 1`] = `
<div
  className="player-songbar-upper-to-play"
  onClick={[Function]}
  onPointerOver={[Function]}
  style={
    Object {
      "height": 35,
    }
  }
/>
`;
